parmed/amber/_amberparm.py

```diff
@@ -1996,6 +1998,31 @@ def _add_missing_13_14(self, ignore_inconsistent_vdw=False):
 
     #===================================================
 
+    def _get_atom_dict_for_alternate_labels(self, residues):
+        from collections import defaultdict
+
+        atom_dict = defaultdict(list)
+        for resid, residue in enumerate(residues):
+            atom_dict[resid] = defaultdict(list)
+            for atom in residue.atoms:
+                atom_dict[resid][atom.name].append(atom)
+        return atom_dict
+
+    def _label_alternates(self, residues=None):
+        # if we know which residue has alternative atoms, we do not need
+        # to iterate all residues
+        if residues is None:
+            residues = self.residues
```

Comment:
What is the use-case for `residues` *not* being `self.residues`?  If there is no use-case, get rid of the option to pass in `residues`.